storage/local/storage.go

```diff
@@ -510,6 +522,11 @@ func (s *memorySeriesStorage) DropMetricsForFingerprints(fps ...clientmodel.Fing
 
 // Append implements Storage.
 func (s *memorySeriesStorage) Append(sample *clientmodel.Sample) {
+	for ln, lv := range sample.Metric {
+		if len(lv) == 0 {
+			delete(sample.Metric, ln)
+		}
+	}
```

Comment:
Mh, I think of it as a global invariant that an empty label value means the label is not set.

The memStorages way of implementing it is to remove these labels from the map – so I consider it more of an implementation detail. For example, for all we know an alternate implementation of `Storage` generates fingerprints that are unaffected by empty labels and thus the respective `SampleAppender` does not have to do this deletion.